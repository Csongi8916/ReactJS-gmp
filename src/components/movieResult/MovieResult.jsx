import React from 'react';
import MovieResultHeader from './MovieResultHeader/MovieResultHeader';
import MovieCard from '../MovieCard/MovieCard';

import classes from './MovieResult.scss';

export default function MovieResult({ isDetailsResult, movies }) {

  const ErrorBoundryTest = () => {
    throw new Error('This is a TEST ERROR!');
  }

  return (
    <div className="MovieResult" style={{ minHeight: '65vh' }}>
      <MovieResultHeader needTotal={!isDetailsResult} needSortToogle={!isDetailsResult} count={movies.length} />
      <div className="MovieResult_MovieCardList">
        {/* ErrorBoundryTest() */}
        {movies && movies.length > 0 ? (
          movies.map(movie => (
            <MovieCard key={movie.id} title={movie.title} image={movie.poster_path} year={movie.release_date} genres={movie.genres} />
          ))
        ) : (
            <p style={{ fontSize: '38px' }}>No films found</p>
          )
        }
      </div>
    </div>
  )
}
