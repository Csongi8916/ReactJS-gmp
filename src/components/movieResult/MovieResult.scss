$bgColor: #424242;

.MovieResult {
  color: #fff;
  background-color: $bgColor;

  .MovieResult_MovieCardList {
    display: flex;
    flex-direction: row;
    justify-content: center;
  }
}

