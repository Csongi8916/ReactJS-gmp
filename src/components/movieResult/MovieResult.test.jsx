import React from 'react';
import renderer from 'react-test-renderer';
import MovieResult from './MovieResult';
import { shallow, mount, render } from 'enzyme';
import { movies } from '../../data/dummy';

describe('MovieResult Snapshpt tests', () => {
  test('Should render correctly MovieResultHeader', () => {
    const tree = renderer.create(<MovieResult isDetailsResult={false} movies={movies} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MovieResult UI tests', () => {
  test('Should number of movies equal number of MovieCard component ', () => {
    const wrapWithData = mount(<MovieResult isDetailsResult={false} movies={movies} />);
    const wrapWithEmpty = mount(<MovieResult isDetailsResult={true} movies={[]} />);

    let movieCardNum = wrapWithData.find('.MovCard').length;
    expect(movies.length).toEqual(movieCardNum);
    movieCardNum = wrapWithEmpty.find('.MovCard').length;
    expect(0).toEqual(movieCardNum);
  });
});