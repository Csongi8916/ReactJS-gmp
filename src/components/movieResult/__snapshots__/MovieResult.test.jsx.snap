// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieResult Snapshpt tests Should render correctly MovieResultHeader 1`] = `
<div
  className="MovieResult"
  style={
    Object {
      "minHeight": "65vh",
    }
  }
>
  <div
    className="MovieResultHeader"
  >
    <div
      className="MovieResultHeader_LeftInfoPanel"
    >
      <span>
        6
         movies found
      </span>
    </div>
    <div
      className="MovieResultHeader_RightInfoPanel"
    >
      <div
        className="NxrToogle"
      >
        <span
          className="NxrToogle_ModeTitle"
        >
          Sort by
        </span>
        <div
          className="NxrToogle_Container"
        >
          <button
            className="NxrToogle_Btn NxrToogle_Btn-Active"
            onClick={[Function]}
          >
            released date
          </button>
          <button
            className="NxrToogle_Btn NxrToogle_Btn-Inactive"
            onClick={[Function]}
          >
            raiting
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MovieResult_MovieCardList"
  >
    <div
      className="MovCard"
    >
      <img
        height="410"
        src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
        width="300"
      />
      <div
        className="Row"
      >
        <span
          className="MovCard_MovieTitle"
        >
          Fifty Shades Freed
        </span>
        <div
          className="MovCard_MovieInfo MovCard_MovieInfoBorder"
        >
          2018
        </div>
      </div>
      <p
        className="MovCard_MovieInfo"
      >
        Drama & Romance
      </p>
    </div>
    <div
      className="MovCard"
    >
      <img
        height="410"
        src="https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg"
        width="300"
      />
      <div
        className="Row"
      >
        <span
          className="MovCard_MovieTitle"
        >
          Zootopia
        </span>
        <div
          className="MovCard_MovieInfo MovCard_MovieInfoBorder"
        >
          2016
        </div>
      </div>
      <p
        className="MovCard_MovieInfo"
      >
        Animation & Adventure & Family & Comedy
      </p>
    </div>
    <div
      className="MovCard"
    >
      <img
        height="410"
        src="https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg"
        width="300"
      />
      <div
        className="Row"
      >
        <span
          className="MovCard_MovieTitle"
        >
          Star Wars: The Last Jedi
        </span>
        <div
          className="MovCard_MovieInfo MovCard_MovieInfoBorder"
        >
          2017
        </div>
      </div>
      <p
        className="MovCard_MovieInfo"
      >
        Fantasy & Adventure & Science Fiction
      </p>
    </div>
    <div
      className="MovCard"
    >
      <img
        height="410"
        src="https://image.tmdb.org/t/p/w500/uxzzxijgPIY7slzFvMotPv8wjKA.jpg"
        width="300"
      />
      <div
        className="Row"
      >
        <span
          className="MovCard_MovieTitle"
        >
          Black Panther
        </span>
        <div
          className="MovCard_MovieInfo MovCard_MovieInfoBorder"
        >
          2018
        </div>
      </div>
      <p
        className="MovCard_MovieInfo"
      >
        Action & Adventure & Fantasy & Science Fiction
      </p>
    </div>
    <div
      className="MovCard"
    >
      <img
        height="410"
        src="https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg"
        width="300"
      />
      <div
        className="Row"
      >
        <span
          className="MovCard_MovieTitle"
        >
          Coco
        </span>
        <div
          className="MovCard_MovieInfo MovCard_MovieInfoBorder"
        >
          2017
        </div>
      </div>
      <p
        className="MovCard_MovieInfo"
      >
        Adventure & Comedy & Family & Animation
      </p>
    </div>
    <div
      className="MovCard"
    >
      <img
        height="410"
        src="https://image.tmdb.org/t/p/w500/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg"
        width="300"
      />
      <div
        className="Row"
      >
        <span
          className="MovCard_MovieTitle"
        >
          Thor: Ragnarok
        </span>
        <div
          className="MovCard_MovieInfo MovCard_MovieInfoBorder"
        >
          2017
        </div>
      </div>
      <p
        className="MovCard_MovieInfo"
      >
        Action & Adventure & Fantasy
      </p>
    </div>
  </div>
</div>
`;
